notebooks/Interpretability - Image Explainers.ipynb

```diff
@@ -0,0 +1,396 @@
+{
+    "cells": [
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Interpretability - Image Explainers\n\nIn this example, we use LIME and Kernel SHAP explainers to explain the ResNet50 model's multi-class output of an image.\n\nFirst we import the packages and define some UDFs and a plotting function we will need later."
+            ],
+            "metadata": {
+                "application/vnd.databricks.v1+cell": {
+                    "title": "",
+                    "showTitle": false,
+                    "inputWidgets": {},
+                    "nuid": "7a2d7f1b-99c4-44d2-9ac5-62ca59002d61"
+                }
+            }
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "from mmlspark.downloader import ModelDownloader\nfrom mmlspark.explainers import *\nfrom mmlspark.cntk import ImageFeaturizer\nfrom mmlspark.stages import UDFTransformer\nfrom pyspark.ml import Pipeline\nfrom pyspark.ml.classification import LogisticRegression\nfrom pyspark.ml.feature import StringIndexer\nfrom pyspark.sql.functions import *\nfrom pyspark.sql.types import *\nimport numpy as np\nimport pyspark\nimport urllib.request\nimport matplotlib.pyplot as plt\nimport PIL, io\nfrom PIL import Image\n\nvec_access = udf(lambda vec, i: float(vec[i]), FloatType())\nvec_slice = udf(lambda vec, indices: (vec.toArray())[indices].tolist(), ArrayType(FloatType()))\narg_top = udf(lambda vec, n: (-vec.toArray()).argsort()[:n].tolist(), ArrayType(IntegerType()))\n\n\ndef plot_superpixels(image_data, sp_clusters, weights):\n    image_bytes = image_data\n    superpixels = sp_clusters\n    green_value = np.percentile(weights, 80)\n    red_value = np.percentile(weights, 20)\n    img = (PIL.Image.open(io.BytesIO(image_bytes))).convert(\"RGBA\")\n    image_array = np.asarray(img).copy()\n    for (sp, v) in zip(superpixels, weights):\n        if v > green_value:\n            for (x, y) in sp:\n                image_array[y, x, 1] = 255\n                image_array[y, x, 3] = 200\n        if v < red_value:\n            for (x, y) in sp:\n                image_array[y, x, 0] = 255\n                image_array[y, x, 3] = 200\n    plt.clf()\n    plt.imshow(image_array)\n    display()"
+            ],
+            "metadata": {
+                "application/vnd.databricks.v1+cell": {
+                    "title": "",
+                    "showTitle": false,
+                    "inputWidgets": {},
+                    "nuid": "ae5c16d1-fd1d-466b-9974-9a514f5f0e92"
+                }
+            },
+            "outputs": [
```

Comment:
I think the outputs needs to be removed before merging?

Reply:
Removed. Thanks!